### PR TITLE
To fix issue#2

### PR DESCRIPTION
**Pull request by lucklove**
fix [won't compile on linux](https://github.com/redox-os/libextra/issues/2)

### CODE REVIEW

File: src/io.rs [hunk 1]
```diff
@@ -20,8 +20,11 @@ impl<W: Write> WriteExt for W {
     }
 
     fn write_char(&mut self, c: char) -> io::Result<usize> {
-        let utf8 = c.encode_utf8();
-        self.write(utf8.as_slice())
+        let mut utf8 = [0; 4];
+        match c.encode_utf8(&mut utf8) {
+            Some(len) => self.write(&utf8[0..len]),
+            None => panic!("<char>.encode_utf8 error!"),    // A buffer of length four is large enough to encode any char
```
Comment (ticki) [inline src/io.rs]: You should instead return a io error with `ErrorKind` `InvalidData`

Comment (lucklove) [inline src/io.rs]: It did return io error, because `panic!` will never be executed(the comment explained)

Comment (ticki) [inline src/io.rs]: Use `unreachable!` then.

Comment (ticki) [inline src/io.rs]: or even `unchecked_unwrap`.

**Comment by ticki** [comment]
Thanks!

File: src/io.rs (hunk 1 above)
Comment (lucklove) [inline src/io.rs]: I think unreachable! is better...

**Comment by lucklove** [comment]
OMG...
Sorry for my careless...
Note that there is already a [PR](https://github.com/redox-os/libextra/pull/1) about this, and it did the opposite thing.
Maybe there should be more tests...

My `rustc` version is `rustc 1.9.0-nightly (02310fd31 2016-03-19)`

Here is the [api](https://doc.rust-lang.org/std/primitive.char.html#method.encode_utf8) I referenced.

**Comment by lucklove** [comment]
My `rustc` version is `rustc 1.9.0-nightly (02310fd31 2016-03-19)`

**Comment by ticki** [comment]
Dammit, it is not you being careless, it is me.

**Comment by ticki** [comment]
I've reverted it.

**Pull request merged**